### ISSUE TITLE
Memoization stopped working on upgrading to 1.0.0

### ISSUE THREAD

**Issue by abime** [svenfuchs/i18n-active_record#128]
We found that Memoization stopped working on upgrading to 1.0.0. 

We believe that the issue got introduced in this #124, where Implementation module got removed and it affected the ancestor chain.

Initially lookup method from Memoize module used to take the precedence but after removal of Implementation module `lookup` method of `I18n::Backend::ActiveRecord` is called. 

### Ancestor Chain in 0.4.1
```ruby
[I18n::Backend::ActiveRecord, I18n::Backend::Memoize, I18n::Backend::ActiveRecord::Implementation, I18n::Backend::Base, I18n::Backend::Transliterator, I18n::Backend::Flatten, ActiveSupport::ToJsonWithActiveSupportEncoder, Object, ActiveAdminHelper, PP::ObjectMixin, ERB::Util, Delayed::MessageSending, ActiveSupport::Dependencies::Loadable, JSON::Ext::Generator::GeneratorMethods::Object, ActiveSupport::Tryable, Kernel, BasicObject]
```

### Ancestor Chain in 1.0.0

```ruby 
[I18n::Backend::ActiveRecord, I18n::Backend::Memoize, I18n::Backend::Flatten, I18n::Backend::Base, I18n::Backend::Transliterator, ActiveSupport::ToJsonWithActiveSupportEncoder, Object, ActiveAdminHelper, PP::ObjectMixin, ERB::Util, Delayed::MessageSending, ActiveSupport::Dependencies::Loadable, JSON::Ext::Generator::GeneratorMethods::Object, ActiveSupport::Tryable, Kernel, BasicObject]
```

Two possible solutions to make memoization work

1. Use `prepend` instead of `include` in the initializer. If we consider taking this approach, we need to make changes to the readme and the initializer generator code.
 ```ruby 
 I18n::Backend::ActiveRecord.send(:prepend, I18n::Backend::Memoize)
 ```
2. Revert removal of `Implementation Module`.

---

Let us know what you think and we can raise the PR accordingly.

Co-Authored by @prayeshshah

**Comment by timfjord** [comment]
Thanks for reporting the issue.

I'd go with the first solution and update the README.
Feel free to open a PR otherwise I will do this next week

**Comment by abime** [comment]
Hey @timfjord 

We have raised a PR for the first solution here #129 

On skimming through code from other i18n backends([simple](https://github.com/ruby-i18n/i18n/blob/3ae91d3ff60aa345a39792feea3c66ef3d8de179/lib/i18n/backend/simple.rb#L22), [key-value](https://github.com/ruby-i18n/i18n/blob/3ae91d3ff60aa345a39792feea3c66ef3d8de179/lib/i18n/backend/key_value.rb#L70)), we noticed that the they all have Implementation module present. And they follow the convention of `including` instead of `prepending` the Memoize module.

Should we also be in sync with other I18n backends and use Implementation module?

**Comment by timfjord** [comment]
Hey @abime 

Thanks for sending the PR

Yeah, I've been thinking about this and I think are right, it is better to stick to the convention and bring the `Implementation` module back.
Sorry for the back and forth.
Please let me know if you wish you update #129 accordingly, otherwise, I will take care of that

**Comment by abime** [comment]
@timfjord No problem. I will update #129 in sometime and send it for review.

**Comment by prayeshshah** [comment]
@timfjord We've made the necessary changes and rebased the branch with master. Please re-review.

**Comment by timfjord** [comment]
Awesome, thanks @prayeshshah 
Let's switch to the PR to discuss the implementation.

**Comment by timfjord** [comment]
Just released [v1.1.0](https://github.com/svenfuchs/i18n-active_record/releases/tag/v1.1.0)

Thanks for the contribution @prayeshshah and @abime
